internal/daemonset/daemonset.go

```diff
@@ -191,7 +198,11 @@ func (dc *daemonSetGenerator) SetDriverContainerAsDesired(ctx context.Context, d
 
 	serviceAccountName := mod.Spec.ModuleLoader.ServiceAccountName
 	if serviceAccountName == "" {
-		serviceAccountName = rbac.GenerateModuleLoaderServiceAccountName(mod)
+		if useDefaultSA {
+			serviceAccountName = "kmm-operator-module-loader"
+		} else {
+			log.FromContext(ctx).Info("Warning: no ServiceAccount set for the ModuleLoader DaemonSet")
```

Comment:
Maybe we should return an error here instead of allowing default namespace service account? It will be clearer for us and user to fail here then later

Reply:
I don't think it should break the flow: users may have configured their RBAC so that the default account has permission to run privileged pods.

Reply:
ok, then let's use util.WarnString API for warning messages

Reply:
> users may have configured their RBAC so that the default account has permission to run privileged pods

I also agree that we have it more flexible and just log a warning. In the future, we could improve the UX by introducing a webhook and/or record an event for example.